bindings/rust/bench/benches/handshake.rs

```diff
@@ -16,19 +14,25 @@ use criterion::{
 pub fn bench_handshake_key_exchange(c: &mut Criterion) {
     fn bench_handshake_for_library<T: TlsBenchHarness>(
         bench_group: &mut BenchmarkGroup<WallTime>,
-        ec_group: &ECGroup,
+        name: &str,
+        ec_group: ECGroup,
     ) {
-        bench_group.bench_function(type_name::<T>(), |b| {
+        // generate all inputs (TlsBenchHarness structs) before benchmarking handshakes
+        // timing only includes negotiation, not config/connection initialization
+        bench_group.bench_function(name, |b| {
             b.iter_batched_ref(
                 || {
                     T::new(&CryptoConfig {
                         cipher_suite: AES_128_GCM_SHA256,
-                        ec_group: *ec_group,
+                        ec_group,
                     })
-                    .unwrap()
                 },
                 |harness| {
-                    harness.handshake().unwrap();
+                    // if harness invalid, do nothing but don't panic
+                    // useful for historical performance bench
```

Comment:
Just curious about this - what is `harness` here and what does it mean for it to be invalid? And why is it useful for the historical benchmarking?

Reply:
`harness` is the benching harness, and it's invalid when the given params (`CryptoConfig`) don't work for that version. I originally did this because as I mentioned above, some configs map to security policies that are invalid for the past versions of s2n-tls. I'll make the comment more descriptive.